pydynamo_brain/pydynamo_brain/ui/radiiPainter.py

```diff
@@ -83,63 +84,63 @@ def drawPoint(self, point, selectedPointID):
             )
             self.maybeDrawText(x, y, point)
 
-    def returnRadiusCoord(self, point, radius):
-        #point radius will be drawn from
+    def returnRadiusCoord(self, point, radius, realRadius):
+        # Point radius will be drawn from
         delta = 1
+        previousPoint = None
+        if not point.isRoot():
+            previousPoint =  point.pathFromRoot()[-2]
+        nextPoint  = point.nextPointInBranch(delta)
 
-        """        #previous point
-        posX, posY, negX, negY = x2, y2, x2, y2,"""
-        if  point.isRoot():
-            #print(point.isRoot)
-            nextPoint  = point.nextPointInBranch(delta)
-            #print(nextPoint)
-            p2Loc = point.location
-            #print(p2Loc)
-            x2, y2, z2 = self.zoomedLocation(p2Loc)
+        pointCoord = point.location
+        xPix, yPix, zPix = self.zoomedLocation(pointCoord)
+        xyPix = np.array([xPix, yPix])
 
-            p1Loc = nextPoint.location
-            x1, y1, z1 = self.zoomedLocation(p1Loc)
+        # Drawing the radius calipers for root points
+        if  previousPoint is  None:
+            pC = self.caliperPoints(point, nextPoint, ifNext= True)
 
+        # Drawing the raidus calipers for terminal points
+        elif nextPoint is None:
+            pC = self.caliperPoints(point, previousPoint, ifNext= False)
 
+        # Drawing calipers for mid branch points, angles based off an average of
+        #Previous and last point
         else:
-            p2Loc = point.location
-            x2, y2, z2 = self.zoomedLocation(p2Loc)
-            point1 =  point.pathFromRoot()[-2:][0]
-            p1Loc = point1.location
-            x1, y1, z1 = self.zoomedLocation(p1Loc)
-
-        #length between two points
-        L = math.sqrt(math.pow((x2-x1),2)+math.pow((y2-y1),2))
-        if L == 0:
-            L = 1
-        #all solutions for the plotting radius
-        #assuming a right triangle
-        negX = x2 -((radius*(y1-y2)/L))
-        posX = x2 +((radius*(y1-y2)/L))
-        posY = y2 +((radius*(x1-x2)/L))
-        negY = y2 -((radius*(x1-x2)/L))
-
+            pA = self.caliperPoints(point, nextPoint, ifNext= True)
+            pB = self.caliperPoints(point, previousPoint, ifNext= False)
+            pC = pA + pB
+
+        pCprime = (radius/np.linalg.norm(pC))*pC
+        radiiPoint = pCprime + xyPix
+        negRadiiPoint = -pCprime + xyPix
+
+        negX = negRadiiPoint[0]
```

Comment:
style: completely optional, and depends on which you find more readable, but i think this could also be:
```
negX, posY = *negRadiiPoint
posX, negY = *radiiPoint
```